common/persistence/sql/class.QueryIterator.php

```diff
@@ -14,99 +14,78 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  *
- * Copyright (c) 2002-2008 (original work) 2014 Open Assessment Technologies SA
+ * Copyright (c) 2002-2020 (original work) 2014 Open Assessment Technologies SA
  *
  */
 
 /**
  * Iterator over all triples
  *
  * @author joel bout <joel@taotesting.com>
- * @package generis
  */
 class common_persistence_sql_QueryIterator implements Iterator
 {
-    const CACHE_SIZE = 100;
-    
+    private const CACHE_SIZE = 100;
+
     /**
      * @var common_persistence_SqlPersistence
      */
     private $persistence;
-    
-    /**
-     * Query to iterator over
-     *
-     * @var string
-     */
+
     private $query;
-    
-    /**
-     * Query parameters
-     *
-     * @var string
-     */
+
     private $params;
-    
+
     /**
      * Id of the current instance
      *
      * @var int
      */
-    private $currentResult = null;
+    private $currentResult;
 
     /**
      * Return statements of the last query
      *
      * @var array
      */
-    private $cache = null;
-    
-    /**
-     * Constructor of the iterator expecting the model ids
-     *
-     * @param array $modelIds
-     */
-    public function __construct(common_persistence_SqlPersistence $persistence, $query, $params = [])
-    {
+    private $cache;
+    private $types;
+    private $limit;
+
+    public function __construct(
+        common_persistence_SqlPersistence $persistence,
+        $query,
+        $params = [],
```

Comment:
```suggestion
        string $query,
        array $params = [],
```

Now we can add type hint :)